vaapisink doesn't keep aspect ratio

Regardless of the options about aspect ration on vaapisink and vaapipostproc, a video played with the iHD codec will stretch to occupy all the output window.

[pipeline.zip](https://github.com/intel/media-driver/files/2321336/pipeline.zip)
here is an example pipeline

@Jister i cloned your whole forked repository but i got a failure to compile

```
/home/phas/Downloads/foo/src/intel-media-driver/media_driver/linux/common/codec/ddi/media_ddi_decode_base.cpp: In member function ‘virtual VAStatus DdiMediaDecode::CreateBuffer(VABufferType, uint32_t, uint32_t, void*, VABufferID*)’:
/home/phas/Downloads/foo/src/intel-media-driver/media_driver/linux/common/codec/ddi/media_ddi_decode_base.cpp:745:14: error: ‘VASubsetsParameterBufferType’ was not declared in this scope
         case VASubsetsParameterBufferType:
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/phas/Downloads/foo/src/intel-media-driver/media_driver/linux/common/codec/ddi/media_ddi_decode_base.cpp:745:14: note: suggested alternative: ‘VASliceParameterBufferType’
         case VASubsetsParameterBufferType:
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
              VASliceParameterBufferType
make[2]: *** [media_driver/CMakeFiles/iHD_drv_video_OBJ.dir/build.make:5276: media_driver/CMakeFiles/iHD_drv_video_OBJ.dir/linux/common/codec/ddi/media_ddi_decode_base.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:170: media_driver/CMakeFiles/iHD_drv_video_OBJ.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

Not sure if this is related to this patch actually

--- a/media_driver/linux/common/ddi/media_libva_putsurface_linux.cpp
+++ b/media_driver/linux/common/ddi/media_libva_putsurface_linux.cpp
@@ -208,8 +208,8 @@ inline Rect_init(
     }
     rect->left                    = destx;
     rect->top                     = desty;
-    rect->right                   = destw;
-    rect->bottom                  = desth;
+    rect->right                   = destx + destw;
+    rect->bottom                  = desty + desth;
 }
 
 VAStatus DdiCodec_PutSurfaceLinuxVphalExt(
@@ -351,10 +351,11 @@ VAStatus DdiCodec_PutSurfaceLinuxHW(
     static VPHAL_SURFACE    Surf;
     VPHAL_SURFACE           target;
     VPHAL_RENDER_PARAMS     renderParams;
+    VPHAL_COLORFILL_PARAMS  colorFill;
 
-    MOS_STATUS              eStatus = MOS_STATUS_INVALID_PARAMETER;
-    RECT                    rect = { 0, 0, 180, 120 };
-    RECT                    dstRect = { 0, 0, 180, 120 };
+    MOS_STATUS              eStatus    = MOS_STATUS_INVALID_PARAMETER;
+    RECT                    srcRect    = { 0, 0, 0, 0 };
+    RECT                    dstRect    = { 0, 0, 0, 0 };
     PDDI_MEDIA_CONTEXT      mediaCtx;
     PDDI_MEDIA_SURFACE      bufferObject;
     uint32_t                width,height,pitch;
@@ -409,15 +410,15 @@ VAStatus DdiCodec_PutSurfaceLinuxHW(
 
     renderParams.Component = COMPONENT_LibVA;
 
-   //Init source rectagle
-    Rect_init(&rect, 0, 0, srcw, srch);
-    Rect_init(&dstRect, dri_drawable->x, dri_drawable->y, dri_drawable->width, dri_drawable->height);
+    //Init source rectangle
+    Rect_init(&srcRect, srcx, srcy, srcw, srch);
+    Rect_init(&dstRect, destx, desty, destw, desth);
 
     // Source Surface Information
-    Surf.Format               = VpGetFormatFromMediaFormat(bufferObject->format);           // Surface format
-    Surf.SurfType             = SURF_IN_PRIMARY;       // Surface type (context)
-    Surf.SampleType           = SAMPLE_PROGRESSIVE;
-    Surf.ScalingMode          = VPHAL_SCALING_AVS;
+    Surf.Format                 = VpGetFormatFromMediaFormat(bufferObject->format);           // Surface format
+    Surf.SurfType               = SURF_IN_PRIMARY;       // Surface type (context)
+    Surf.SampleType             = SAMPLE_PROGRESSIVE;
+    Surf.ScalingMode            = VPHAL_SCALING_AVS;
 
     Surf.OsResource.Format      = VpGetFormatFromMediaFormat(bufferObject->format);
     Surf.OsResource.iWidth      = bufferObject->iWidth;
@@ -429,10 +430,14 @@ VAStatus DdiCodec_PutSurfaceLinuxHW(
     Surf.OsResource.bo          = bufferObject->bo;
     Surf.OsResource.pGmmResInfo = bufferObject->pGmmResourceInfo;
 
-    Surf.ColorSpace            = DdiVp_GetColorSpaceFromMediaFormat(bufferObject->format);
-    Surf.ExtendedGamut         = false;
-    Surf.rcSrc                 = rect;
-    Surf.rcDst                 = dstRect;
+    Surf.dwWidth                = bufferObject->iWidth;
+    Surf.dwHeight               = bufferObject->iHeight;
+    Surf.dwPitch                = bufferObject->iPitch;
+    Surf.TileType               = VpGetTileTypeFromMediaTileType(bufferObject->TileType);
+    Surf.ColorSpace             = DdiVp_GetColorSpaceFromMediaFormat(bufferObject->format);
+    Surf.ExtendedGamut          = false;
+    Surf.rcSrc                  = srcRect;
+    Surf.rcDst                  = dstRect;
 
     MOS_LINUX_BO* drawable_bo = mos_bo_gem_create_from_name(mediaCtx->pDrmBufMgr, "rendering buffer", buffer->dri2.name);
     if  (nullptr == drawable_bo)
@@ -475,7 +480,7 @@ VAStatus DdiCodec_PutSurfaceLinuxHW(
     target.SurfType              = SURF_OUT_RENDERTARGET;
 
     //init target retangle
-    Rect_init(&rect, 0, 0, dri_drawable->width, dri_drawable->height);
+    Rect_init(&srcRect, dri_drawable->x, dri_drawable->y, dri_drawable->width, dri_drawable->height);
     Rect_init(&dstRect, dri_drawable->x, dri_drawable->y, dri_drawable->width, dri_drawable->height);
 
     // Create GmmResourceInfo
@@ -501,18 +506,23 @@ VAStatus DdiCodec_PutSurfaceLinuxHW(
     target.OsResource.bo         = drawable_bo;
     target.OsResource.pData      = (uint8_t *)drawable_bo->virt;
     target.OsResource.TileType   = tileType;
+    target.TileType              = tileType;
     target.dwWidth               = dri_drawable->width;
     target.dwHeight              = dri_drawable->height;
     target.dwPitch               = target.OsResource.iPitch;
     target.ColorSpace            = CSpace_sRGB;
     target.ExtendedGamut         = false;
-    target.rcSrc                 = rect;
+    target.rcSrc                 = srcRect;
     target.rcDst                 = dstRect;
 
-    renderParams.uSrcCount          = 1;
-    renderParams.uDstCount          = 1;
-    renderParams.pSrc[0]            = &Surf;
-    renderParams.pTarget[0]         = &target;
+    renderParams.uSrcCount                  = 1;
+    renderParams.uDstCount                  = 1;
+    renderParams.pSrc[0]                    = &Surf;
+    renderParams.pTarget[0]                 = &target;
+    renderParams.pColorFillParams           = &colorFill;
+    renderParams.pColorFillParams->Color    = 0xFF000000;
+    renderParams.pColorFillParams->bYCbCr   = false;
+    renderParams.pColorFillParams->CSpace   = CSpace_sRGB;
 
     DdiMediaUtil_LockMutex(&mediaCtx->PutSurfaceRenderMutex);
     eStatus = vpHal->Render(&renderParams);

